libs/core/algorithms/tests/unit/algorithms/detail/chunk_size_idx_iterator.cpp

```diff
@@ -0,0 +1,176 @@
+//  Copyright (c) 2021 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/iterator_support/tests/iterator_tests.hpp>
+#include <hpx/modules/datastructures.hpp>
+#include <hpx/modules/testing.hpp>
+#include <hpx/parallel/util/detail/chunk_size_iterator.hpp>
+
+#include <algorithm>
+#include <cstddef>
+#include <iterator>
+#include <list>
+#include <numeric>
+#include <utility>
+#include <vector>
+
+// define operators for the type returned by the chunk_size_iterator to make
+// the tests compile
+template <typename Iter>
+bool operator==(hpx::tuple<Iter, std::size_t, std::size_t> const& lhs,
+    hpx::tuple<Iter, std::size_t> const& rhs)
+{
+    return hpx::get<0>(lhs) == hpx::get<0>(rhs) &&
+        hpx::get<1>(lhs) == hpx::get<1>(rhs) &&
+        hpx::get<2>(lhs) == hpx::get<2>(rhs);
+}
+
+// Special tests for RandomAccess Iterator.
+template <typename Iterator, typename Value>
+void category_test(
+    Iterator start, Iterator finish, Value, std::random_access_iterator_tag)
+{
+    using difference_type =
+        typename std::iterator_traits<Iterator>::difference_type;
+
+    difference_type distance = std::distance(start, finish);
+
+    // Pick a random position internal to the range
+    difference_type offset = (unsigned) rand() % distance;
+
+    HPX_TEST(offset >= 0);
+
+    Iterator internal = start;
+    std::advance(internal, offset);
+
+    // Try some binary searches on the range to show that it's ordered
+    HPX_TEST(std::binary_search(start, finish, *internal));
+
+    std::pair<Iterator, Iterator> xy(
+        std::equal_range(start, finish, *internal));
+    Iterator x = xy.first, y = xy.second;
+
+    HPX_TEST(std::distance(x, y) == 1);
+
+// disable warning: unary minus operator applied to unsigned type, result
+// still unsigned
```

Comment:
What triggers this warning and why can it be ignored?

Reply:
No idea, this is copied straight from the Boost iterator test suite - which reminds me that I need to add their copyright...

Reply:
Could you give it a try without ignoring the warning? I suppose it might be a warning inside the MSVC `std::binary_search` implementation :/

Reply:
I have removed the #pragmas now as I don't see the warning locally. It might have been that Boost was using these functions in a different context, causing the warning to show up.